packages/devtools_app/lib/src/service/service_manager.dart

```diff
@@ -496,6 +498,36 @@ class ServiceConnectionManager {
   }
 }
 
+class ResolvedUriManager {
+  final _resolvedUrlMap = <String, String?>{};
+  final _unknownUris = <String>[];
+  VmServiceWrapper? service;
+
+  Future<void> fetchUnknownUris(
+    String isolateId,
+  ) async {
+    final packageUris = (await serviceManager.service!
+            .lookupPackageUris(isolateId, _unknownUris))
+        .uris;
+    if (packageUris != null) {
+      for (var i = 0; i < _unknownUris.length; i++) {
+        final unknownUri = _unknownUris[i];
+        final resolvedUri = packageUris[i];
+        _resolvedUrlMap[unknownUri] = resolvedUri!;
+      }
+      _unknownUris.clear();
+    }
+  }
+
+  String? getPackageUri(String uri) {
```

Comment:
avoid starting method names with get: https://dart.dev/guides/language/effective-dart/design#avoid-starting-a-method-name-with-get

rename to `lookupPackageUri`